src/Driver/RemoteClient.php

```diff
@@ -396,7 +396,7 @@ private function parse(?string $data = null): void
             }
         } catch (\Throwable $exception) {
             // Parser *should not* throw an exception, but in case it does...
-            $this->logger->critical($exception);
+            $this->logger->critical($exception->getMessage(), ['exception' => $exception]);
```

Comment:
This should probably cast the exception to a string?

Reply:
I don't think so. I checked Symfony to see how they're doing it and it turns out that they're not even using the exception message. They always use some string literal for instance:

```php
$this->logger->error('An exception was thrown when handling an AccessDeniedException.', array('exception' => $e));
```

So it would actually be better to not even use the message.

Reply:
In that case this shouldn't use the message, but rather say something like the comment above says: `'Unexpected ' . \get_class($e) . ' while parsing data from client @ ' . (string) $this->clientAddress`

Reply:
See https://github.com/amphp/http-server/pull/306#discussion_r403727293. Gimme message suggestions for the individual logger calls below if you don't like mine.